R/initRepo.R

```diff
@@ -1,24 +1,38 @@
 
-##' This helper function create a new repository, creates and checks
-##' out \sQuote{gh-pages} branch and fills it with the required path.
-##' 
-##' Currently only \sQuote{src/contrib} for source repositories is
-##' supported.
+##' This helper function create a new repository, creates and checks out
+##' \sQuote{gh-pages} branch and fills it with the required path for both source
+##' and binary repositories. 
 ##'
-##' This function is still undergoing development and polish and may
-##' change in subsequent versions.
+##' This function is still undergoing development and polish and may change in
+##' subsequent versions.
 ##'
 ##' @title Intialize a git repo for drat
-##' @param name A character variable with the name the new repository,
-##' the default is \dQuote{drat}.
-##' @param basepath A character variable with path to the directory in
-##' which the new repository is to be created. The default value is
-##' \dQuote{"~/git"}.
-##' @return The function is invoked for its side-effects and only
-##' returns \code{NULL} invisibly.
+##' @param name A character variable with the name the new repository, the
+##'   default is \dQuote{drat}. If name has two parts in format of
+##'   \dQuote{account_name/repo_name}, the cooresponding github repo is set as
+##'   repo git remote.
+##' @param basepath A character variable with path to the directory in which the
+##'   new repository is to be created. The default value is \dQuote{~/git}.
+##' @param branch The github repo branch to serve the repo. Default to be 
+##' \code{gh-pages}.
+##' @param remote_type Add github remote with \dQuote{SSH} or \dQuote{HTTPS}
+##'   format.
+##' @param push Push changes to github remote in the end if available. This may
+##'   only works for SSH and when ssh-agent is available.
+##' @return The function is invoked for its side-effects and only returns
+##'   \code{NULL} invisibly.
 ##' @author Dirk Eddelbuettel
-initRepo <- function(name="drat", basepath="~/git") {
-
+initRepo <- function(name="drat", basepath="~/git", 
```

Comment:
instead of making all the parts individual arguments and pasteing them together, why not just have the user pass in a url?

Reply:
I think it's easy to just copy `username/repo` from github url, and that should be enough for a lot of user cases. In most cases the branch doesn't need to changed from default too.